cli/src/commands/parallelize.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Write;
+use std::rc::Rc;
+
+use indexmap::IndexSet;
+use itertools::Itertools;
+use jj_lib::backend::CommitId;
+use jj_lib::commit::Commit;
+use jj_lib::repo::Repo;
+use jj_lib::revset::{RevsetExpression, RevsetIteratorExt};
+use tracing::instrument;
+
+use crate::cli_util::{short_commit_hash, CommandHelper, RevisionArg};
+use crate::command_error::{user_error, CommandError};
+use crate::commands::rebase::rebase_descendants;
+use crate::ui::Ui;
+
+/// Parallelize revisions by making them siblings
+///
+/// The set of target commits being parallelized must have a single head and
+/// root commit (not to be confused with the repo root), otherwise the command
+/// will fail. Each of the target commits is rebased onto the parents of the
+/// root commit. The children of the head commit are rebased onto the target
+/// commits.
+///
+/// Example usage:
+/// ```text
+/// jj log
+/// ◉   3
+/// ◉   2
+/// ◉   1
+/// ◉
+/// jj parallelize 1::2
+/// jj log
+/// ◉     3
+/// ├─╮
+/// │ ◉   2
+/// ◉ │   1
+/// ├─╯
+/// ◉
+/// ```
```

Comment:
Minor nit,: use the same graph style as we have in other commands.

Reply:
I changed the comment to match the other style. But, I don't think that typing out a bunch of commit graphs by hand is a great use of time. If we want to add more examples, is there a way to generate this style automatically with `jj log`?

Reply:
there used to be the other be the ascii log, but that got removed.

Reply:
Another option is to update all other graphs with the new style, if you prefer.

Reply:
> Another option is to update all other graphs with the new style, if you prefer.

Haha, maybe another day. But we probably should do that eventually since users will be used to seeing that graph format.